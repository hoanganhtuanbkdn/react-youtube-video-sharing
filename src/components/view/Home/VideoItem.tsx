import VideoPlayer from '@/components/ui/VideoPlayer';
import { ISharing } from '@/services';
import { DeleteOutlined } from '@ant-design/icons';
import React, { memo } from 'react';

import { Popconfirm } from 'antd';

export const VideoItem = ({
	item,
	onDeleteSharing,
}: {
	item: ISharing;
	onDeleteSharing: ((id: number) => Promise<void>) | undefined;
}) => {
	return (
		<div className="flex flex-col overflow-hidden rounded-md border border-gray-300 shadow md:h-[290px] md:flex-row">
			<div className="relative flex h-[200px] w-full md:h-full md:w-[440px]">
				<VideoPlayer
					videoUrl={item.videoUrl}
					cover={item.cover || ''}
					videoKey={item.id!}
				/>
			</div>
			<div className="col flex-1 items-start justify-start gap-1 px-5 py-3 text-xs md:gap-3 md:text-base">
				<h1 className="!my-0 text-xl md:text-2xl">{item.title}</h1>
				<p className="!my-0">
					<strong>Share by:</strong> {item.user?.email}
				</p>
				<p className="!my-0 font-bold">Description</p>
				<p className="!my-0 line-clamp-3">{item.description}</p>
				{!!onDeleteSharing && (
					<div className="flex w-full items-end justify-end">
						<Popconfirm
							title="Delete the item"
							description="Are you sure to delete this item?"
							okText="Yes"
							cancelText="No"
							onConfirm={() => onDeleteSharing?.(item.id!)}
						>
							<DeleteOutlined style={{ color: 'red' }} />
						</Popconfirm>
					</div>
				)}
			</div>
		</div>
	);
};

export default memo(VideoItem);
