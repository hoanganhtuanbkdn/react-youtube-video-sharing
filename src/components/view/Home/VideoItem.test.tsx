import { render } from '@testing-library/react';
import React from 'react';

import VideoItem from './VideoItem';

// Must be imported before the tested file
Object.defineProperty(window, 'matchMedia', {
	writable: true,
	value: jest.fn().mockImplementation((query) => ({
		matches: false,
		media: query,
		onchange: null,
		addListener: jest.fn(), // deprecated
		removeListener: jest.fn(), // deprecated
		addEventListener: jest.fn(),
		removeEventListener: jest.fn(),
		dispatchEvent: jest.fn(),
	})),
});

const mockData: any = {
	id: 85,
	videoUrl: 'https://www.youtube.com/watch?v=pvBXT229quc',
	cover: 'https://i.ytimg.com/vi/pvBXT229quc/maxresdefault.jpg?sqp=-oaymwEmCIAKENAF8quKqQMa8AEB-AH-CYAC0AWKAgwIABABGFQgZSglMA8=&rs=AOn4CLBNRtBUDTmOqUHqHa5P8V8Ymy6okg',
	title: 'Làm Thế Nào Để Vượt Qua Giới Hạn Của Bản Thân (Rất Hay) - Thầy Minh Niệm',
	description:
		'Làm Thế Nào Để Vượt Qua Giới Hạn Của Bản Thân (Rất Hay) - Thầy Minh Niệm',
	createdAt: null,
	userId: 2,
	user: {
		id: 2,
		email: 'tuanpha.it@gmail.com',
		firstName: 'Tuan',
		lastName: 'Pham',
		roles: ['admin'],
		branchId: null,
		resetKey: null,
		resetCount: null,
		resetTimestamp: null,
		resetKeyTimestamp: null,
		createdAt: '2023-05-26T07:03:17.235Z',
		updatedAt: null,
	},
};

describe('rendering and submitting share form', () => {
	test('renders ShareForm correctly', () => {
		const handleClick: any = jest.fn((id: number) => {
			// console.log('values', values);
		});
		const { container } = render(
			<VideoItem onDeleteSharing={handleClick} item={mockData} />
		);

		expect(container.firstChild).toMatchSnapshot();
	});
});
