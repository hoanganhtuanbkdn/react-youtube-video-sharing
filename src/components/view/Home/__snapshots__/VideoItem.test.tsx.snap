// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering and submitting share form renders ShareForm correctly 1`] = `
<div
  class="flex flex-col overflow-hidden rounded-md border border-gray-300 shadow md:h-[290px] md:flex-row"
>
  <div
    class="relative flex h-[200px] w-full md:h-full md:w-[440px]"
  >
    <div
      class="relative max-h-full w-full  overflow-hidden bg-cover bg-center bg-no-repeat"
      style="background-image: url(https://i.ytimg.com/vi/pvBXT229quc/maxresdefault.jpg?sqp=-oaymwEmCIAKENAF8quKqQMa8AEB-AH-CYAC0AWKAgwIABABGFQgZSglMA8=&rs=AOn4CLBNRtBUDTmOqUHqHa5P8V8Ymy6okg);"
    >
      <div
        class="center animation absolute inset-0 z-20 overflow-hidden bg-cover bg-center bg-no-repeat duration-200 opacity-100"
        style="background-image: url(https://i.ytimg.com/vi/pvBXT229quc/maxresdefault.jpg?sqp=-oaymwEmCIAKENAF8quKqQMa8AEB-AH-CYAC0AWKAgwIABABGFQgZSglMA8=&rs=AOn4CLBNRtBUDTmOqUHqHa5P8V8Ymy6okg);"
      >
        <button
          class="cursor-pointer border-none bg-transparent p-4"
          title="jGooooo - click me"
        >
          <span
            aria-label="play-circle"
            class="anticon anticon-play-circle"
            role="img"
            style="color: white; font-size: 40px;"
          >
            <svg
              aria-hidden="true"
              data-icon="play-circle"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
              />
              <path
                d="M719.4 499.1l-296.1-215A15.9 15.9 0 00398 297v430c0 13.1 14.8 20.5 25.3 12.9l296.1-215a15.9 15.9 0 000-25.8zm-257.6 134V390.9L628.5 512 461.8 633.1z"
              />
            </svg>
          </span>
        </button>
      </div>
      <div
        class="z-10"
        style="width: 100%; height: 100%;"
      />
    </div>
  </div>
  <div
    class="col flex-1 items-start justify-start gap-1 px-5 py-3 text-xs md:gap-3 md:text-base"
  >
    <h1
      class="!my-0 text-xl md:text-2xl"
    >
      Làm Thế Nào Để Vượt Qua Giới Hạn Của Bản Thân (Rất Hay) - Thầy Minh Niệm
    </h1>
    <p
      class="!my-0"
    >
      <strong>
        Share by:
      </strong>
       
      tuanpha.it@gmail.com
    </p>
    <p
      class="!my-0 font-bold"
    >
      Description
    </p>
    <p
      class="!my-0 line-clamp-3"
    >
      Làm Thế Nào Để Vượt Qua Giới Hạn Của Bản Thân (Rất Hay) - Thầy Minh Niệm
    </p>
    <div
      class="flex w-full items-end justify-end"
    >
      <span
        aria-label="delete"
        class="anticon anticon-delete"
        role="img"
        style="color: red;"
        tabindex="-1"
      >
        <svg
          aria-hidden="true"
          data-icon="delete"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M360 184h-8c4.4 0 8-3.6 8-8v8h304v-8c0 4.4 3.6 8 8 8h-8v72h72v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80h72v-72zm504 72H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM731.3 840H292.7l-24.2-512h487l-24.2 512z"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;
